src/data/nat/choose/multinomial.lean

```diff
@@ -0,0 +1,171 @@
+/-
+Copyright (c) 2022 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+Heavily inspired by code from Kyle Miller and Kevin Buzzard
+-/
+import algebra.big_operators.basic
+import algebra.big_operators.fin
+import data.nat.choose.basic
+import data.finset.basic
+import data.list.perm
+import data.fin.vec_notation
+
+import tactic.linarith
+
+/-!
+# Multinomial
+
+This file defines the multinomial coefficient and several small lemma's for manipulating it.
+
+## Main declarations
+
+- `nat.multinomial`: the multinomial coefficient
+
+-/
+
+open_locale nat
+
+namespace nat
+
+/-- The multinomial coefficient. Gives the number of strings consisting of symbols
+from `s`, where `c ∈ s` appears with multiplicity `f c`.
+
+Defined as `(s.sum f)! / s.prod (factorial ∘ f)`
+-/
+def multinomial {α} (s : finset α) (f : α → ℕ) : ℕ := (s.sum f)! / s.prod (factorial ∘ f)
```

Comment:
If you `open_locale big_operators` you can use the ∑ ∏ notation, which feels nicer; same below.
You may declare `variables {α : Type*} (s : finset α) (f : α → ℕ)` and remove them from statements below if you want. Not so much to make statements shorter as they're already short, but to save you some work to correct the spacing everywhere ...